### PR TITLE
Issue #390: Use namedtuple instead of tuple for Hash_ShortHash_Message, BranchDef, ReflogEntry

### LINKED ISSUE #390: Use namedtuple instead of tuple in `Hash_ShortHash_Message = Tuple[str, str, str]`
I know that type name: `Hash_ShortHash_Message` has field names in it but it gets lost along the way and it can be easier and cleaner to use namedtuple.
e.g.

change this:
`        

        commits: List[Hash_ShortHash_Message] = self.__git.get_commits_between(
            opt_fork_point, current_branch)

        if not commits:
            raise MacheteException(
                "No commits to squash. Use `-f` or `--fork-point` to specify the "
                "start of range of commits to squash.")

        if len(commits) == 1:
            sha, short_sha, subject = commits[0]
            print(f"Exactly one commit ({short_sha}) to squash, ignoring.\n")
            print("Tip: use `-f` or `--fork-point` to specify where the range of "
                  "commits to squash starts.")
            return`

to this:
`        

        if len(commits) == 1:
            print(f"Exactly one commit ({commits[0].short_sha}) to squash, ignoring.\n")
            print("Tip: use `-f` or `--fork-point` to specify where the range of "
                  "commits to squash starts.")
            return`

### PR DESCRIPTION
**Pull request by amalota**
closes https://github.com/VirtusLab/git-machete/issues/390

### CODE REVIEW

**Comment by PawelLipski** [comment]
Hmmm so this `namedtuple` feature is supposed to work only since 3.8... how come do the CI tests for 3.6 and 3.7 pass? 😯

**Comment by amalota** [comment]
@PawelLipski It looks like it is available in python 3.6 https://docs.python.org/3.6/search.html?q=namedtuple&check_keywords=yes&area=default

**Comment by PawelLipski** [comment]
Hmmm okay... so where did you take the information about it only being available in Python 3.8. then from? 🤔

**Comment by amalota** [comment]
Here https://docs.python.org/3/library/typing.html#typing.NamedTuple, above `Other special directives` there is a text: `New in version 3.8.` but as it turns out its for feature above it, not below as i initially thought.

File: git_machete/client.py [hunk 1]
```diff
@@ -1472,7 +1465,7 @@ def generate_entries() -> Generator[Tuple[FullCommitHash, BranchDef], None, None
                     self.__branch_defs_by_sha_in_reflog[sha] = [branch_def]
 
             def log_result() -> Generator[str, None, None]:
-                branch_defs_: List[BranchDef]
+                branch_defs_: List[BranchPair]
```
Comment (PawelLipski) [inline git_machete/client.py]: Pls rename all vars/def like this one adequately

File: git_machete/git_operations.py [hunk 2]
```diff
@@ -149,7 +149,20 @@ def __init__(self, to_hash: FullCommitHash, while_descendant_of_hash: FullCommit
...
+
+
+class GitReflogEntry(NamedTuple):
+    hash: FullCommitHash
+    reflog_subject: str
+
+
+class BranchPair(NamedTuple):
+    local_branch: LocalBranchShortName
+    remote_branch: AnyBranchName
```
Comment (PawelLipski) [inline git_machete/git_operations.py]: `local_or_remote_branch` (see e.g. `def branch_def_to_str`)

**Review by PawelLipski — APPROVED** [review]

**Pull request merged**